cdap-formats/src/main/java/co/cask/cdap/format/StructuredRecordStringConverter.java

```diff
@@ -66,13 +64,12 @@ public static StructuredRecord fromJsonString(String json, Schema schema) throws
    * Converts a {@link StructuredRecord} to a delimited string.
    */
   public static String toDelimitedString(final StructuredRecord record, String delimiter) {
-    return Joiner.on(delimiter).join(
-      Iterables.transform(record.getSchema().getFields(), new Function<Schema.Field, String>() {
-        @Override
-        public String apply(Schema.Field field) {
-          return record.get(field.getName()).toString();
-        }
-      }));
+    return String.join(delimiter, record.getSchema().getFields().stream()
+      .map(field -> {
+        Object val = record.get(field.getName());
+        return val == null ? "" : val.toString();
+      })
+      .collect(Collectors.toList()));
```

Comment:
There is a collector that do string joining.

```java
    return record.getSchema().getFields().stream()
      .map(Schema.Field::getName)
      .map(record::get)
      .map(value -> value == null ? "" : value.toString())
      .collect(Collectors.joining(delimiter));
```

Reply:
nice, updated